packages/photon/src/generation/TSClient.ts

```diff
@@ -432,30 +433,36 @@ export class Photon {
         this.generator ? JSON.stringify(this.generator) : 'undefined'
       },
       showColors: this.errorFormat === 'pretty'
-    })
+    }
+
+    this.engine = new Engine(this.engineConfig)
 
     this.dmmf = new DMMFClass(dmmf)
 
-    this.fetcher = new PhotonFetcher(this, this.engine, false, internal.hooks, this.errorFormat)
+    this.fetcher = new PhotonFetcher(this, false, internal.hooks)
   }
-  private async connectEngine(publicCall?: boolean) {
-    return this.engine.start()
-  }
-  connect(): Promise<void> {
+  async connect(): Promise<void> {
+    if (this.disconnectionPromise) {
+      await this.disconnectionPromise
+    }
     if (this.connectionPromise) {
       return this.connectionPromise
     }
-    this.connectionPromise = this.connectEngine(true)
+    this.connectionPromise = this.engine.start()
     return this.connectionPromise!
   }
   async disconnect() {
-    await this.engine.stop()
+    if (this.connectionPromise) {
+      await this.connectionPromise
```

Comment:
Is there a timeout on the connectionPromise?

Reply:
Yes, in `NodeEngine`!

Reply:
Yes, in `NodeEngine`!